x/valset-pref/simulation/sim_msgs.go

```diff
@@ -172,10 +172,7 @@ func GetRandomValAndWeights(ctx sdk.Context, k valsetkeeper.Keeper, sim *osmosim
 			return nil, fmt.Errorf("No validator")
 		}
 
-		randValue, err := RandomWeight(remainingWeight)
-		if err != nil {
-			return nil, fmt.Errorf("Error with random weights")
-		}
+		randValue := sdk.MustNewDecFromStr(fmt.Sprintf("%.2f", sim.RandomDecAmount(remainingWeight)))
```

Comment:
Similat to another comment, I think this should be fine but please confirm before committing
```suggestion
		randValue := sim.RandomDecAmount(remainingWeight)
```

Reply:
Confirmed this is working by changing the other suggestion and rerunning the simulator